File: examples/java/hds_prog_halfedge_iterator.java
import CGAL.HalfedgeDS.HalfedgeDS;
import CGAL.HalfedgeDS.HalfedgeDS_decorator;
import CGAL.HalfedgeDS.HDS_Halfedge_handle;

public class hds_prog_halfedge_iterator {
  public static void main(String arg[]){
    HalfedgeDS hds = new HalfedgeDS();
    HalfedgeDS_decorator decorator = new HalfedgeDS_decorator(hds);
    decorator.create_loop();
    decorator.create_segment();
    assert decorator.is_valid();
    
    int n = 0;
    for ( HDS_Halfedge_handle h : hds.halfedges() ){ ++n; }
    assert n == 4;
  }
}

File: examples/python/hds_prog_halfedge_iterator.py
from CGAL.CGAL_HalfedgeDS import HalfedgeDS
from CGAL.CGAL_HalfedgeDS import HalfedgeDS_decorator
from CGAL.CGAL_HalfedgeDS import HDS_Halfedge_handle


hds = HalfedgeDS()
decorator = HalfedgeDS_decorator(hds)
decorator.create_loop()
decorator.create_segment()
assert decorator.is_valid()

n = 0
for h in hds.halfedges():
  n+=1

assert n == 4
